modules/postgresql/main.tf

```diff
@@ -44,7 +44,7 @@ locals {
   // Force the usage of connector_enforcement
   connector_enforcement = var.connector_enforcement ? "REQUIRED" : "NOT_REQUIRED"
 
-  database_name = var.enable_default_db ? google_sql_database.default[0].name : (length(local.databases) > 0 ? google_sql_database.additional_databases[0].name : "")
+  database_name = var.enable_default_db ? var.db_name : (length(var.additional_databases) > 0 ? var.additional_databases[0].name : "")
```

Comment:
@q2w this will only show the name of first database. Whats the purpose of this database name in output?

```
var.additional_databases[0].name
```

Reply:
database_name is part of an output called env_vars which has other database related details. The env_vars can be used by cloud-run, MIG or other modules directly. Eg. https://github.com/q2w/solution-builder-three-tier-web-app/blob/feature/ttwa-adc-default-network-sa-in-cr/infra/main.tf#L39 

```
output "env_vars" {
  description = "Exported environment variables"
  value = {
    "CLOUD_SQL_DATABASE_HOST" : google_sql_database_instance.default.first_ip_address,
    "CLOUD_SQL_DATABASE_CONNECTION_NAME" : google_sql_database_instance.default.connection_name,
    "CLOUD_SQL_DATABASE_NAME" : local.database_name
  }
}
```

https://github.com/terraform-google-modules/terraform-google-sql-db/blob/master/modules/postgresql/outputs.tf#L152

Reply:
@bharathkkb

Reply:
@q2w thx. But your logic is showing only one database name in case of multiple additional databases. Is that acceptable behavior?

Reply:
Yes, We provide one database name (default if present, first additional database otherwise) as part of env_vars. Thus saving users to worry about database name in their application.
Providing all databases in the env_vars doesn't help users as they still need to configure which database to use in the application (which can be done directly).